### DISCUSSION TITLE
Remove comitup-WiFi-credentials

### DISCUSSION THREAD

**Discussion by outdoorbits** [outdoorbits/little-backup-box#193] [Announcements]
Thank to an idea of [rkosick](https://github.com/rkosick) there is a new option in setting: Remove comitup-WiFi-credentials.

**Comment by qdmt** [comment]
Great update. As an FYI - I don't see the setting, and the automatic installation of comitup seemed to have gone away. I've reimaged the Raspberry OS Lite 64 from scratch as well as tried to install comitup manually. I might have just done something wrong along the way - I am due to try again from scratch once I downgrade the internal SD from 128 to 32gb.

**Comment by outdoorbits** [comment]
Just tested a new installation, works as expected.

**Reply by rkosick** [reply]
I just did the update and works well. Great job. It can be useful for troubleshooting and testing at home.

I'm still wondering if it would be possible, to make a setting to execute arbitrary commands in case someone would need it. Like this implemented feature, only I can choose the command. Would be nice not to need to connect via a ssh app on the phone.

**Reply by qdmt** [reply]
I started again from a clean install on a new stick, and confirming all worked as well. I might have fudged the re-image process initially :). Thanks again @outdoorbits.

**Comment by outdoorbits** [comment]
I consider the option to execute freely worded command line jobs to be 
dangerous. lbb should stand for the best possible data security. Of 
course, freely formulated commands are relatively error-prone.
Technically, commands would also have to be passed on several times 
within the system and ultimately executed. In my opinion, the risk that 
any special characters within a command would lead to a 
misinterpretation and modification of the command cannot be ruled out.
I would be much more responsible for careful extensions of the tools if 
necessary. And we should possibly discuss such.

**Reply by rkosick** [reply]
Yeah that's a good point.

**Comment by rkosick** [comment]
Since I was testing around with the new wireguard function I noticed something regarding the option to remove comitup-WiFi-credentials that may be interesting and I didn't now before:
The command apparently only removes the active connection. If more than one connection is saved, it would still connect to one that was not active when you remove the credentials.
One thing I would call a bug right now:  I also tried to remove credentials while beeing connected to to comitup AP 10.41.01.1.  I was just curious what would happen. This made lbb not show "FINISHED" after the command was executed. Resulting to not be available anymore at all. AP was not showing up anymore. Had to pull the power and was lucky that the SD card was not corrupted. After restart comitup AP was available again.

**Comment by outdoorbits** [comment]
Did you parallel connect to VPN?

**Reply by rkosick** [reply]
I don't think so. I deleted the VPN configuration file and wanted to start all over again so I wanted to delete the credentials also. And before, VPN was only active after initiate the backup. Not sure if you changed that. But I am pretty sure it was not active. But I can test it again on the weekend.

**Comment by outdoorbits** [comment]
Please do so. VPN actually is still disabled after backup as before.

**Reply by rkosick** [reply]
Since I was connected to the comitup AP and had no active connection to an active wifi connection, VPN shouldn't been active to? There was one saved connection but it was not active. And I didn't execute any backup. Did this just after reboot. But I can test it again on the weekend. It's not a big deal if you know it. Normally you wouldn't try to deleted the credentials while not connected to a network.

**Comment by outdoorbits** [comment]
Hard to understand now. But we will find out, hopefully...

**Reply by outdoorbits** [reply]
Could you reproduce the issue? I would like to remove the warning about alpha status.

**Reply by rkosick** [reply]
Sorry I was busy. You can remove the warning about alpha status. It has nothing to do with VPN, that works great.
The only thing I could reproduce is the problem I described. If you are not connected to a known network and use the option to delete the comitup credentials (while connected to the comitup AP), LBB is not responding any more. AP is gone. After pulling the power plug it and starting lbb again the AP was there again.
As I said, there is no reason to do something like that. But if you do it yo may break the system while performing a hard reset by cutting power.

**Comment by outdoorbits** [comment]
After removing the comitup credentials it starts in hotspot mode. What type is your connection in the moment before you remove the credentials?

**Reply by rkosick** [reply]
Like I said: hotspot mode.

**Reply by rkosick** [reply]
When connected to a known WIFI it works like it should.

**Comment by outdoorbits** [comment]
So I think this is just about comitup. I will remove alpha status. Thank you.

**Reply by rkosick** [reply]
I agreee. Great job.
